lib/PINNBench/DiffusionReaction/hpo.py

```diff
@@ -0,0 +1,116 @@
+import os
+import torch
+import numpy as np
+from pdebench.models.pinn.train import *
+from .model import FNN
+from deephyper.problem import HpProblem
+from deephyper.search.hps import CBO
+from deephyper.evaluator import profile, RunningJob
+from deephyper.stopper.integration import DeepXDEStopperCallback
+from deephyper.stopper import LCModelStopper
+from deephyper_benchmark.utils.json_utils import array_to_json
+from fvcore.nn import FlopCountAnalysis
+from deephyper_benchmark.integration.torch import count_params
+
+
+@profile
+def run(job: RunningJob) -> dict:
+    config = job.parameters
+    dataset = os.environ.get('DEEPHYPER_BENCHMARK_DATASET')
+
+    stopper_callback = DeepXDEStopperCallback(job)
+
+    val_loss, test_loss, losshistory, model = run_training(
+        net_class=FNN,
+        scenario="diff-react",
+        epochs=config["epochs"],
+        learning_rate=config["lr"],
+        model_update=500,
+        root_path="./DiffusionReaction/build/PDEBench-DH/pdebench/data/" + dataset,
+        flnm=dataset + ".h5",
+        config=config,
+        seed="0000",
+        callbacks=stopper_callback,
+    )
+    param_count = count_params(model)
+    flops = FlopCountAnalysis(model, inputs=(torch.randn(1, 3))).total()
+
+    train_ls = np.array(losshistory.loss_train).sum(axis=1)
+    val_ls = np.array(losshistory.loss_test).sum(axis=1)
+    steps = np.array(losshistory.steps)
+    lc_train_X = np.stack([steps, train_ls], axis=1)
+    lc_val_X = np.stack([steps, val_ls], axis=1)
+    lc_train_X_json = array_to_json(lc_train_X)
+    lc_val_X_json = array_to_json(lc_val_X)
+
+    objective = -val_loss
+    metadata = {
+        "num_hyperparameters": len(job.parameters),
+        "num_parameters":param_count['num_parameters'],
+        "num_parameters_train":param_count["num_parameters_train"],
+        "val_loss": val_loss,
+        "test_metrics": array_to_json(test_loss),
```

Comment:
how many metrics/elements do you have in this array? can we stick to the format `train_X, val_X, test_X` ?

Reply:
Following the PDEBench paper, there are 6 metrics for testing scores. But I can change it to a single value just to keep it consistent.

Reply:
`X` should be replaced by the name of the metric, and you should have 1 key per metric in the dictionary, for example `train_mse, train_r2, train_mae, valid_mse, valid_r2, valid_mae...`